src/main/java/reactor/kafka/receiver/internals/ConsumerEventLoop.java

```diff
@@ -266,22 +270,31 @@ public void run() {
                                 toResume.removeAll(this.pausedByUser);
                                 this.pausedByUser.clear();
                                 consumer.resume(toResume);
+                                isPaused.set(false);
```

Comment:
at each point `isPaused` is set to a boolean, `pausedByUs` has already been set to that same boolean. I'm not saying I'm sure this is a suitable replacement, but wondering if there's a good reason not to use `pausedByUs`? The only thing I can think of is that `pausedByUs` is set to `true` in more cases than the ones for which we want to `consumer.wakeup()` above at line 142.

Reply:
Yes; we can use `pausedByUs` by pushing it up to the enclosing class.

However, I don't believe we can consider this PR until the CLA is "signed".

Reply:
Good point, made changes to use `pausedByUs` as indicator to wake up the consumer.
Signed CLA as well.